### PR TITLE
Upgrade to latest Megatron-lm version

### PR DESCRIPTION
**Pull request by ericharper**
There were some small breaking changes when using the latest Megatron-lm version.

Fixing those here.

# TODO
~~We need to cut a new pip package from Megatron and update NeMo requirements.~~

### CODE REVIEW

File: nemo/collections/nlp/models/nlp_model.py [hunk 1]
```diff
@@ -353,9 +353,10 @@ def setup(self, stage: str) -> None:
                     raise NotImplementedError(
                         f'The BERT encoder: {self.bert_model} does not support model parallelism yet.'
                     )
-            else:
-                # Megatron without model parallelism
-                self.complete_megatron_init()
+            # else:
+            #  This is getting called before DDP is initialized
+            #     # Megatron without model parallelism
+            #     self.complete_megatron_init()
```
Comment (ekmb) [inline nemo/collections/nlp/models/nlp_model.py:359]: Is this needed?

File: nemo/collections/nlp/modules/common/megatron/megatron_bert.py [hunk 2]
```diff
@@ -78,13 +83,28 @@ def __init__(self, model_name, config, vocab_file, model_parallel_size=None, mod
...
         else:
-            extra_args_provider = None
+            # megatron expects batch size is not None
+            def _update_megatron_args(parser):
+                parser.set_defaults(micro_batch_size=1)
+                parser.set_defaults(scaled_masked_softmax_fusion=False)
+                parser.set_defaults(bias_gelu_fusion=False)
+                parser.set_defaults(bias_dropout_fusion=False)
+
+                return parser
```
Comment (ekmb) [inline nemo/collections/nlp/modules/common/megatron/megatron_bert.py:101]: Could you please combine two _update_megatron_args() methods and add an extra param `model_parallel_size`

**Review by ekmb — CHANGES_REQUESTED** [review]

File: nemo/collections/nlp/models/nlp_model.py (hunk 1 above)
Comment (ericharper) [reply nemo/collections/nlp/models/nlp_model.py]: Not needed. Removed.

File: nemo/collections/nlp/modules/common/megatron/megatron_bert.py (hunk 2 above)
Comment (ericharper) [reply nemo/collections/nlp/modules/common/megatron/megatron_bert.py]: Done.

**Review by ekmb — APPROVED** [review]

**Pull request merged**